pkg/endpoint/bpf.go

```diff
@@ -407,25 +408,64 @@ func updateCT(owner Owner, e *Endpoint, epIPs []net.IP,
 	return wg
 }
 
-// allocateRedirects must be called while holding the endpoint and consumable
+// addNewRedirectsFromMap must be called while holding the endpoint and consumable
 // locks for writing. On success, returns nil; otherwise, returns an error
 // indicating the problem that occurred while adding an l7 redirect for the
 // specified policy.
-func (e *Endpoint) allocateRedirects(owner Owner, m policy.L4PolicyMap) error {
+func (e *Endpoint) addNewRedirectsFromMap(owner Owner, m policy.L4PolicyMap, requiredRedirects map[string]struct{}) error {
```

Comment:
I know it's stated in the comments on the Endpoint type but could you add which locks should be held for these functions?

Reply:
Done.